packages/split-button/src/Menu/Menu.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { ComponentType, ReactElement } from 'react';
+import isEmpty from 'lodash/isEmpty';
+
+import Button from '@leafygreen-ui/button';
+import { cx } from '@leafygreen-ui/emotion';
+import Icon from '@leafygreen-ui/icon';
+import { useDarkMode } from '@leafygreen-ui/leafygreen-provider';
+import { isComponentType } from '@leafygreen-ui/lib';
+import { Menu as LGMenu, MenuItemProps } from '@leafygreen-ui/menu';
+
+import {
+  triggerBaseStyles,
+  triggerSizeStyles,
+  triggerThemeStyles,
+} from './Menu.styles';
+import { MenuProps } from './Menu.types';
+
+/**
+ * @internal
+ */
+export const Menu = ({
+  variant,
+  disabled,
+  align,
+  justify,
+  size,
+  baseFontSize,
+  menuItems,
+  containerRef,
+  id,
+  ...rest
+}: MenuProps) => {
+  const { theme } = useDarkMode();
+
+  const renderMenuItems = () => {
+    if (menuItems) {
+      if ('props' in menuItems) {
+        if (!isEmpty(menuItems.props)) {
+          // Array of menuItems
+          if (Array.isArray(menuItems.props.children)) {
+            return menuItems.props.children.map(
+              (
+                menuItem: ComponentType<MenuItemProps>,
+                index: number,
+              ): ReactElement | undefined => {
+                if (menuItem == null) {
+                  return menuItem;
+                }
+
+                if (isComponentType(menuItem, 'MenuItem')) {
+                  return React.cloneElement(menuItem, {
+                    active: false,
+                    key: `menuItem-${index}`,
+                  });
+                } else {
+                  console.warn('Please use a LeafyGreen <MenuItem> component.');
+                }
+              },
+            );
+          }
+
+          // Object with one menuItem
+          // React returns an object instead of an array if there is one item
+          const menuItem = menuItems.props.children;
+
+          if (isComponentType(menuItem, 'MenuItem')) {
+            return React.cloneElement(menuItem, {
+              active: false,
+            });
+          } else {
+            console.warn('Please use a LeafyGreen <MenuItem> component.');
+          }
+        }
+      }
+    }
+  };
+
+  return (
+    <LGMenu
+      data-testid="lg-split-button-menu"
+      align={align}
+      justify={justify}
+      refEl={containerRef}
+      trigger={
+        <Button
```

Comment:
(if need be, we can turn this into its own component and use this: 
```
useEffect(() => {
    let unregisterRipple: (() => void) | undefined;
    const backgroundColor = rippleColors[theme][variant];

    if (rippleRef.current != null && !disabled) {
      unregisterRipple = registerRipple(rippleRef.current, {
        backgroundColor,
      });
    }

    return unregisterRipple;
  }, [rippleRef, variant, darkMode, disabled, theme]);
```
 to register a ripple

Reply:
Just checked and it looks like the ripple never gets properly deployed when a Button component is passed to the trigger prop of a Menu.

Reply:
oddly, it works for <Tooltip />

Reply:
talked in person, we have a ticket and i will try to not use the trigger prop